compare.py

```diff
@@ -0,0 +1,413 @@
+    # save the output image
+    diff_png = "{}{}{}".format(folder, os.sep, "diff.png")
+    logging.debug("Diff image saved to {}".format(diff_png))
+
+    cv2.imwrite(diff_png, latest_image)
+    before_and_after_png = "{}{}{}".format(folder, os.sep, "before_and_after.png")
+
+    logging.info("\tBefore and after png saved to: {}".format(before_and_after_png))
+    cv2.imwrite(before_and_after_png, tile_images(left_image=datum_image, right_image=latest_image))
+
+    return before_and_after_png
+
+
+def save_text(folder, text, filename="diff.txt"):
+
+    path = "{}\{}".format(folder, filename)
+    f = codecs.open(path, 'w', encoding='utf8')
+
+    for text in text:
+        logging.debug("Extracted the following text from: [{}]".format(text))
+        f.write(u'\n=======================================================================\n')
+        f.write(text)
+
+    f.flush()
+    f.close()
+
+    return path
+
+
+def extract_regions_of_interest(contours, gray_datum_image, gray_latest_image, threshold=0.75,
+                                contour_area_threshold=100):
+    count = 0
+    rectangles = []
+
+    logging.debug("Processing {} contours to process".format(len(contours)))
+    for contour in contours:
+        count = count + 1
+        logging.debug("Processing contour {} of {}".format(count, len(contours)))
+
+        # compute the bounding box of the contour for the region of interest
+        (x, y, w, h) = cv2.boundingRect(contour)
+        region_of_interest = gray_latest_image[y:y + h, x:x + w]
+
+        # extract text from region of interest
+        text_of_interest = pytesseract.image_to_string(region_of_interest)
+
+        if len(text_of_interest) <= 0:
+            logging.debug("No text, so let's skip this contour.....")
+            continue
+
+        logging.debug("Found text: {}".format(text_of_interest))
+
+        # search for the region of interest in the original image
+        loc = skimage.feature.match_template(gray_datum_image, region_of_interest, pad_input=False,
+                                             mode='constant', constant_values=0)
+
+        match = np.where(loc >= threshold)
+        logging.debug("correlation factor: {}".format(np.sum(match) / (loc.shape[0] * loc.shape[1])))
+
+        if np.sum(match) / (loc.shape[0] * loc.shape[1]) <= 0 and h * w > contour_area_threshold:
+            rectangles.append([x, y, x + w, y + h])
+
+    return rectangles
+
+
+def preprocess_images(folder, datum_image, latest_image):
+
+    datum_image, latest_image = load_images(folder=folder, datum_image=datum_image, latest_image=latest_image)
+    datum_image, latest_image = make_image_size_identical(datum_image=datum_image, latest_image=latest_image)
+
+    # Convert the images to gray scale
+    gray_datum_image = cv2.cvtColor(datum_image, cv2.COLOR_BGR2GRAY)
+    gray_latest_image = cv2.cvtColor(latest_image, cv2.COLOR_BGR2GRAY)
+
+    return datum_image, gray_datum_image, latest_image, gray_latest_image
```

Comment:
this is better `return (datum_image, gray_datum_image, latest_image, gray_latest_image)`